app/src/main/java/science/apolline/view/activity/MainActivity.kt

```diff
@@ -227,28 +217,38 @@ class MainActivity : RootActivity(), NavigationView.OnNavigationItemSelectedList
         INFLUXDB_SYNC_FREQ = (mPrefs.getString("sync_frequency", "60")).toLong()
         // Launch AutoSync
         SyncJobScheduler.setAutoSync(SYNC_MOD, INFLUXDB_SYNC_FREQ, this)
-        if (mPrefs.getString("sensor_name" , "sensor_name does not exist").toLowerCase().contains(regex = SplashScreen.IOIO_REGEX)) {
+        if (mPrefs.getString("sensor_name", "sensor_name does not exist").toLowerCase().contains(regex = SplashScreen.IOIO_REGEX)) {
             supportActionBar!!.setBackgroundDrawable(ColorDrawable(Color.parseColor("#ffdc41")))
             var bundle = Bundle()
-            bundle.putString("sensor_name" , mPrefs.getString("sensor_name" , "sensor_name does not exist"))
+            bundle.putString("sensor_name", mPrefs.getString("sensor_name", "sensor_name does not exist"))
             mFragmentViewPager.setArguments(bundle)
             startService(Intent(applicationContext, IOIOService::class.java))
             replaceFragment(mFragmentViewPager)
         }
+        if (savedInstanceState == null) {
+            //APPA
+            if (mPrefs.getString("sensor_name", "sensor_name does not exist").toLowerCase().contains(regex = SplashScreen.APPA_REGEX)) {
+                supportActionBar!!.setBackgroundDrawable(ColorDrawable(Color.parseColor("#428aff")))
+                val gattServiceIntent = Intent(this@MainActivity, BluetoothLeService::class.java)
+                bindService(gattServiceIntent, mServiceConnection, Context.BIND_AUTO_CREATE)
+            }
+            SYNC_MOD = (mPrefs.getString("sync_mod", "2")).toInt()
+            INFLUXDB_SYNC_FREQ = (mPrefs.getString("sync_frequency", "60")).toLong()
+        }
     }
 
-    override fun onStart() {
+    /*override fun onStart() {
         super.onStart()
 
         //APPA
-        if (mPrefs.getString("sensor_name" , "sensor_name does not exist").toLowerCase().contains(regex = SplashScreen.APPA_REGEX)) {
+        if (mPrefs.getString("sensor_name", "sensor_name does not exist").toLowerCase().contains(regex = SplashScreen.APPA_REGEX)) {
             supportActionBar!!.setBackgroundDrawable(ColorDrawable(Color.parseColor("#428aff")))
             val gattServiceIntent = Intent(this@MainActivity, BluetoothLeService::class.java)
             bindService(gattServiceIntent, mServiceConnection, Context.BIND_AUTO_CREATE)
         }
         SYNC_MOD = (mPrefs.getString("sync_mod", "2")).toInt()
         INFLUXDB_SYNC_FREQ = (mPrefs.getString("sync_frequency", "60")).toLong()
-    }
+    }*/
```

Comment:
Why this code is commented ? Should it be removed or not ?

Reply:
Not sure, this was already there before, I can remove it